File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/decorator/dataexport/JsonDataExportDecorator.java
// JsonDataExportDecorator.java

package com.bigboxcode.designpattern.decorator.dataexport;

public class JsonDataExportDecorator extends DataExportDecorator {
    public JsonDataExportDecorator(DataExport dataExporter) {
        super(dataExporter);
    }

    @Override
    public void processData() {
        dataExporter.processData();
        this.processJson();
    }

    private void processJson() {
        System.out.println("Processed data to JSON");
    }
}

File: design-patterns/ts/decorator/data-export/json-data-export-decorator.ts
// json-data-export-decorator.ts

import DataExport from "./data-export";
import DataExportDecorator from "./data-export-decorator";


class JsonDataExportDecorator extends DataExportDecorator {
    constructor(dataExporter: DataExport) {
        super(dataExporter);
    }

    processData(): void {
        this.dataExporter.processData();
        this.processJson();
    }

    processJson(): void {
        console.log("Processed data to JSON");
    }
}

export default JsonDataExportDecorator;